### PR TITLE
Refactoring parts of code

### PR DESCRIPTION
**Pull request by AliceBalfanz**
This PR: 
- Splits testing and creating summary outputs into two parts
- adds bash script needed for automation of test runs and updating testing environment
- uploading first results for manual test run

Missing and not implemented yet:
- comparing test results to previous test run and ending email upon changes
- still more refactoring possible, not finished yet

Relevant files for this pr: 
testing-cci-datasets/auto-test-cate-datasets.sh
testing-cci-datasets/generate_summary_and_send_email.py
testing-cci-datasets/test_cci_data_support.py
testing-cci-datasets/update_cate_envs.sh

### CODE REVIEW

File: testing-cci-datasets/auto-test-cate-datasets.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+test_directory=~/projects/cate-e2e/testing-cci-datasets
+project_dirs=~/projects
+declare -a test_modes=("development" "stage" "production")
+
+git pull
+# you need to specify here, which envs should be rebuild. For stage and production this only
+# happens if a new dev or stable release has been issued. After building a new env for stage and production
+# please remember to remove them from update_envs array
+#declare -a update_envs=("development")
+declare -a update_envs=("dev_local")
```
Comment (AliceBalfanz) [inline testing-cci-datasets/auto-test-cate-datasets.sh:17]: This is just a placeholder. Not updating development env currently due to issue with newest dask version

File: testing-cci-datasets/auto-test-cate-datasets.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+source ~/miniconda3/bin/activate cate-env-production
+for cci_store in "${cci_stores[@]}";do
+  python generate_summary_and_send_email.py "$cci_store" "$test_mode"
+done
+source ~/miniconda3/bin/deactivate
+
+git add .
+git commit -m "automatic update of test results"
+git push
```
Comment (AliceBalfanz) [inline testing-cci-datasets/auto-test-cate-datasets.sh:43]:
```suggestion
git push

```

**Review by TonioF — APPROVED** [review]
Two comments, neither of which should block from merging.

File: testing-cci-datasets/auto-test-cate-datasets.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+set -e
+set -m
+
+# export necessary environment variables
+export EMAIL_PASSWORD="rxH6BDDy4UqSs2k"
```
[the remark this answers is not in the document]
Comment (AliceBalfanz) [reply testing-cci-datasets/auto-test-cate-datasets.sh:6]: Yes, we have to solve this differently. Or we add the auto-test-cate-dataset.sh to our service-ke repo.

File: testing-cci-datasets/auto-test-cate-datasets.sh [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  echo "activated environment $cate_env"
+  for cci_store in "${cci_stores[@]}";do
+    python "$test_directory"/test_cci_data_support.py "$cci_store" "$test_mode"
+  done
+  source ~/miniconda3/bin/deactivate
+done
+
+source ~/miniconda3/bin/activate cate-env-production
+for cci_store in "${cci_stores[@]}";do
+  python generate_summary_and_send_email.py "$cci_store" "$test_mode"
```
[the remark this answers is not in the document]
Comment (AliceBalfanz) [reply testing-cci-datasets/auto-test-cate-datasets.sh:37]: yes, you are very correct, I'll rename the script (it does not have the functionality to send emails yet anyway) and yes, it should loop over test modes. Fixing right away

File: testing-cci-datasets/auto-test-cate-datasets.sh [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  done
+done
+source ~/miniconda3/bin/deactivate
+
+git add .
+git commit -m "automatic update of test results"
+git push
+
+# writing automated email still needs to be implemened
+#   python write_email.py
```
Comment (AliceBalfanz) [inline testing-cci-datasets/auto-test-cate-datasets.sh:47]:
```suggestion
#   python write_email.py

```

**Pull request merged**